<a href="https://colab.research.google.com/github/yunhatsal/py/blob/main/CTBC_9.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1. 두가지 이상의 선택, 다방향 선택구조 코딩
- 배송료 처리 3가지로 코딩
- 변수 : price
- 조건 : 10만원 이상은 배송비 0, 2만원 이상은 3천원, 2만원 이하는 5천원

In [ ]:
price = int(input("구입금액 입력: "))         #변수 price 사용자가 정수형 금액 입력
if price >= 100000:                           # price의 값이 10만원보다 크거나 같으면
  shipping_cost = 0                           # 선택1. 배송료(shipping_cost)는 0원
else:                                         # 그게 아니면
  if price >= 20000:                          # 선택2. price값이 2만원보다 크거나 같으면
    shipping_cost = 3000                      # 배송료(shipping_cost)는 3000원
  else:                                       # 그것도 아니면
    shipping_cost = 5000                      # 선택3. 배송료(shipping_cost)는 5000원
print(f'배송료는 {shipping_cost}원 입니다.')  # F-String문자 Format으로 최종 배송료 출력


구입금액 입력: 25000
배송료는 3000원 입니다.


위 코드는 사용자로부터 구입금액을 입력받아서 배송료를 계산하는 코드입니다. 구입금액에 따라서 배송료가 다르게 책정됩니다.

만약 구입금액이 100,000원 이상이라면 배송료는 0원입니다.
구입금액이 20,000원 이상이고 100,000원 미만이라면 배송료는 3,000원입니다.
그 외의 경우에는 배송료가 5,000원으로 책정됩니다.
최종 배송료를 출력합니다

2. 반복구조를 이용한 "환영합니다."를 5번 출력하는 코드작성
- 변수 : Welcome
- 출력내용 : 환영합니다.를 세로로 5번, 10번, 30번


In [ ]:
print("환영합니다.")  # "환영합니다."라는 메시지를 출력합니다.
print("환영합니다.")  # "환영합니다."라는 메시지를 출력합니다.
print("환영합니다.")  # "환영합니다."라는 메시지를 출력합니다.
print("환영합니다.")  # "환영합니다."라는 메시지를 출력합니다.
print("환영합니다.")  # "환영합니다."라는 메시지를 출력합니다.

위 코드는 "환영합니다."라는 메시지를 각 줄마다 출력하는 코드입니다.

In [ ]:
print("환영합니다.\n" *5) # "환영합니다."라는 메시지를 5번 반복하여 출력합니다.

환영합니다.
환영합니다.
환영합니다.
환영합니다.
환영합니다.



위 코드는 "환영합니다."라는 메시지를 5번 출력하는 코드입니다. * 연산자를 사용하여 문자열을 반복 출력합니다

In [ ]:
count = 1  # 초기 카운트 변수를 1로 설정합니다.
while count <= 5:  # count가 5 이하인 동안 반복합니다.
  print("(", count, ")", "환영합니다. ")  # count와 "환영합니다." 메시지를 출력합니다.
  count = count + 1  # count 값을 1씩 증가시킵니다.

( 1 ) 환영합니다. 
( 2 ) 환영합니다. 
( 3 ) 환영합니다. 
( 4 ) 환영합니다. 
( 5 ) 환영합니다. 


위 코드는 초기 count 변수를 1로 설정하고, while 반복문을 사용하여 count가 5 이하인 동안 반복합니다. 각 반복마다 count와 "환영합니다." 메시지를 출력하고, count 값을 1씩 증가시킵니다.

3. 0부터 9까지 출력하는 코드 작성
- 변수 : x

In [ ]:
x = 0  # 변수 x를 0으로 초기화합니다.
while x < 10:  # x가 10보다 작은 동안 반복합니다.
  #print(x)  # x 값을 출력합니다. (주석 처리된 부분)
  print(x, end=' ')  # x 값을 출력하고, 줄바꿈 대신 공백으로 끝을 설정합니다.
  x = x + 1  # x 값을 1씩 증가시킵니다.

0 1 2 3 4 5 6 7 8 9 

위 코드는 변수 x를 0으로 초기화하고, while 반복문을 사용하여 x가 10보다 작은 동안 반복합니다. 각 반복마다 x 값을 출력하고, 줄바꿈 대신 공백으로 끝을 설정하여 출력합니다. 그리고 x 값을 1씩 증가시킵니다.

In [ ]:
x = 0  # 변수 x를 0으로 초기화합니다.
while x <= 9:  # x가 9 이하인 동안 반복합니다.
  print(x, end=' ')  # x 값을 출력하고, 줄바꿈 대신 공백으로 끝을 설정합니다.
  x += 1  # x 값을 1씩 증가시킵니다.

0 1 2 3 4 5 6 7 8 9 

위 코드는 변수 x를 0으로 초기화하고, while 반복문을 사용하여 x가 9 이하인 동안 반복합니다. 각 반복마다 x 값을 출력하고, 줄바꿈 대신 공백으로 끝을 설정하여 출력합니다. 그리고 x 값을 1씩 증가시킵니다.

4. 1부터 10까지 합산하는 코드작성

In [ ]:
x = 1  # 변수 x를 1로 초기화합니다.
total = 0  # 변수 total을 0으로 초기화합니다.
while x <= 10:  # x가 10 이하인 동안 반복합니다.
  total += x  # total에 x 값을 더합니다.
  #total = total + 1  # (주석 처리된 부분) total에 1을 더합니다.
  #x = x + 1  # (주석 처리된 부분) x 값을 1씩 증가시킵니다.
  x += 1  # x 값을 1씩 증가시킵니다.
print(total)  # total 값을 출력합니다.

55


위 코드는 변수 x를 1로 초기화하고, 변수 total을 0으로 초기화합니다. while 반복문을 사용하여 x가 10 이하인 동안 반복합니다. 각 반복마다 total에 x 값을 더합니다. 주석 처리된 부분은 total과 x 값을 각각 1씩 증가시키는 부분이며, 현재 주석 처리되어 있습니다. 마지막으로 total 값을 출력합니다.

5. 3-6-9 게임(3의 배수 판단) : 3의 배수를 3으로 나누면 나머지가 0이 되므로(i mod 3)의 값이 0이면 박수를 출력, 아니면 숫자 출력하는 코드

In [ ]:
for num in range(1, 21):  # 1부터 20까지의 숫자에 대해 반복합니다.
  if num % 3 == 0:  # num이 3으로 나누어 떨어지는지 확인합니다.
    print("박수", end=' ')  # 3으로 나누어 떨어지면 "박수"를 출력합니다.
  else:
    print(num, end=' ')  # 3으로 나누어 떨어지지 않으면 num을 출력합니다.

1 2 박수 4 5 박수 7 8 박수 10 11 박수 13 14 박수 16 17 박수 19 20 

위 코드는 1부터 20까지의 숫자에 대해 반복하는 for 반복문을 사용합니다. 각 숫자에 대해 if 문을 사용하여 숫자가 3으로 나누어 떨어지는지 확인합니다. 만약 3으로 나누어 떨어지면 "박수"를 출력하고, 나머지 경우에는 숫자를 출력합니다. 출력 시 줄바꿈 대신 공백으로 끝을 설정합니다

In [ ]:
num = 1  # 변수 num을 1로 초기화합니다.
while num <= 20:  # num이 20보다 작거나 같은 동안 반복합니다.
    if num % 3 == 0:  # 만약 num을 3으로 나눈 나머지가 0이라면 (num이 3의 배수라면)
        print("박수", end=' ')  # "박수"를 출력합니다. end=' '는 줄바꿈 없이 공백으로 분리해서 출력하게 합니다.
    else:
        print(num, end=' ')  # 그 외의 경우에는 num을 출력합니다.
    num += 1  # num을 1씩 증가시킵니다.

1 2 박수 4 5 박수 7 8 박수 10 11 박수 13 14 박수 16 17 박수 19 20 

이 코드는 1부터 20까지의 숫자를 출력하면서, 숫자가 3의 배수인 경우에는 대신 "박수"를 출력하는 예제입니다. 코드를 간단히 설명하면 다음과 같습니다:

먼저 num이라는 변수를 1로 초기화합니다.
while문을 사용하여 num이 20보다 작거나 같은 동안 반복합니다.
if문을 사용하여 num을 3으로 나눈 나머지가 0인지 확인합니다. 만약 0이라면 (즉, num이 3의 배수라면), "박수"를 출력합니다.
그렇지 않은 경우에는 num을 출력합니다.
마지막으로 num을 1씩 증가시킵니다.
따라서 이 코드를 실행하면 1부터 20까지의 숫자를 출력하면서 3의 배수인 경우에는 대신 "박수"를 출력하게 됩니다.

6. 팩토리얼 계산하는 코드 작성
- 1! = 1, 2!=2(1)=2, 3!=3(2)(1)=6

In [ ]:
i = 5  # 변수 i를 5로 초기화합니다.
factorial = 1  # 변수 factorial을 1로 초기화합니다.
while i >= 1:  # i가 1 이상인 동안 반복합니다.
    factorial *= i  # factorial에 i를 곱하여 factorial 값을 업데이트합니다.
    i -= 1  # i를 1씩 감소시킵니다.
print(factorial)  # 최종적인 factorial 값을 출력합니다.

120


이 코드는 5부터 1까지의 숫자를 이용하여 팩토리얼 값을 계산하는 예제입니다. 코드를 간단히 설명하면 다음과 같습니다:

먼저 i라는 변수를 5로 초기화합니다.
factorial이라는 변수를 1로 초기화합니다. 팩토리얼 값을 저장할 변수입니다.
while문을 사용하여 i가 1 이상인 동안 반복합니다.
반복문 내부에서 factorial에 i를 곱하여 factorial 값을 업데이트합니다.
i를 1씩 감소시킵니다.
반복이 끝나면 최종적인 factorial 값을 출력합니다.
따라서 이 코드를 실행하면 factorial 변수에는 5부터 1까지의 숫자를 모두 곱한 결과인 120이 저장되며, 이 값이 출력됩니다.

In [ ]:
# 계산할 숫자를 입력받습니다.
num = int(input("팩토리얼을 계산할 숫자를 입력하시오.: "))

# 초기값으로 1을 설정합니다.
factorial = 1

# 팩토리얼 계산을 위한 반복문인 while입니다.
while num >= 1:                 # num값이 1보다 작거나 같을 때 하위 문장(코드)을 진행
  factorial = factorial * num   # 현재 num값을 factorial값에 곱해줍니다.
  num -= 1                      # num값을 1씩 감소시킵니다.

print(f'계산된 팩토리얼 값은 {factorial} 입니다.') # 계산된 팩토리얼 값을 출력합니다.

팩토리얼을 계산할 숫자를 입력하시오.: 5
계산된 팩토리얼 값은 120 입니다.


이 코드는 사용자로부터 입력받은 숫자의 팩토리얼 값을 계산하는 코드입니다.

먼저, input() 함수를 사용하여 사용자로부터 숫자를 입력받습니다. 입력받은 값은 int() 함수를 사용하여 정수형으로 변환합니다.

다음으로, factorial 변수를 1로 초기화합니다. 이 변수는 팩토리얼 값을 저장할 변수입니다.

그 후, while 반복문을 사용하여 num 값이 1보다 크거나 같을 때까지 아래의 코드를 반복합니다.

안쪽의 코드는 현재 num 값을 factorial 값에 곱해주고, num 값을 1씩 감소시킵니다. 이렇게 하면 num 값이 1씩 감소하면서 팩토리얼 값을 계산할 수 있습니다.

마지막으로, 계산된 팩토리얼 값을 출력합니다.

이렇게 코드를 작성하면 사용자로부터 입력받은 숫자의 팩토리얼 값을 계산하고 출력할 수 있습니다.

7. 구구단 출력해주는 코드 작성
- 변수 : dan
- 반복문(1) : for문 사용
- 반복문(2) : while문 사용

In [ ]:
# for문 사용
dan = int(input("구구단에서 몇 단을 출력할까요: "))  # 사용자로부터 출력할 단을 입력받습니다.
for i in range(1, 10):
    print(dan, "x", i, "=", dan * i)  # 구구단 계산식을 출력합니다.

구구단에서 몇 단을 출력할까요: 4
4 x 1 = 4
4 x 2 = 8
4 x 3 = 12
4 x 4 = 16
4 x 5 = 20
4 x 6 = 24
4 x 7 = 28
4 x 8 = 32
4 x 9 = 36


이 코드는 사용자로부터 입력받은 숫자에 해당하는 구구단을 출력하는 코드입니다.

먼저, input() 함수를 사용하여 사용자로부터 출력할 단을 입력받습니다. 입력받은 값은 int() 함수를 사용하여 정수형으로 변환합니다.

그 후, for 반복문을 사용하여 1부터 9까지의 범위를 순회합니다. range(1, 10)은 1부터 9까지의 숫자를 나타냅니다.

반복문 내부에서는 print() 함수를 사용하여 구구단 계산식을 출력합니다. dan은 사용자가 입력한 단의 숫자이며, i는 반복문에서 순회하는 변수입니다. 따라서 dan * i는 구구단의 계산 결과를 나타냅니다.

이렇게 코드를 작성하면 사용자가 입력한 숫자에 해당하는 구구단을 출력할 수 있습니다.

In [ ]:
# while문 사용
dan = int(input("구구단에서 몇 단을 출력할래?: "))  # 사용자로부터 출력할 단을 입력받습니다.
i = 1  # 초기값으로 1을 설정합니다.
while i <= 9:  # i가 9보다 작거나 같을 때까지 반복합니다.
    print(dan, "x", i, "=", dan * i)  # 구구단 계산식을 출력합니다.
    i += 1  # i 값을 1씩 증가시킵니다.

구구단에서 몇 단을 출력할래?: 4
4 x 1 = 4
4 x 2 = 8
4 x 3 = 12
4 x 4 = 16
4 x 5 = 20
4 x 6 = 24
4 x 7 = 28
4 x 8 = 32
4 x 9 = 36


이 코드는 사용자로부터 입력받은 숫자에 해당하는 구구단을 출력하는 코드입니다.

먼저, input() 함수를 사용하여 사용자로부터 출력할 단을 입력받습니다. 입력받은 값은 int() 함수를 사용하여 정수형으로 변환합니다.

다음으로, i 변수를 1로 초기화합니다. 이 변수는 반복문에서 순회하는 변수로서, 구구단의 곱셈을 위해 사용됩니다.

while 반복문은 i가 9보다 작거나 같을 때까지 반복합니다. 구구단은 1부터 9까지의 숫자를 곱하는 것이므로, 1부터 9까지의 범위를 순회하도록 설정합니다.

반복문 내부에서는 print() 함수를 사용하여 구구단 계산식을 출력합니다. dan은 사용자가 입력한 단의 숫자이며, i는 반복문에서 순회하는 변수입니다. 따라서 dan * i는 구구단의 계산 결과를 나타냅니다.

마지막으로, i 값을 1씩 증가시켜줍니다. 이렇게 하면 다음 순회 때 다음 숫자의 구구단을 출력할 수 있습니다.

이렇게 코드를 작성하면 사용자가 입력한 숫자에 해당하는 구구단을 출력할 수 있습니다.